### PR TITLE
CONSOLE-3183: Improve console.dashboards/overview/detail/item extension

### PR DESCRIPTION
**Pull request by jcaianirh**
Address: https://issues.redhat.com/browse/CONSOLE-3183

Deprecate the old OverviewDetailItem extension and replace it with CustomOverviewDetailItem extension. The previous extension was not opinionated enough in that it didn't require the developer to customize the dashboard with the OverviewDetailItem console component. The new extension renders its contents as children of the OverviewDetailItem component so all spacing and css elements should match up correctly and render properly.

Thanks to @andrewballantyne and @florkbr for the help with this one!

### CODE REVIEW

File: dynamic-demo-plugin/locales/en/plugin__console-demo-plugin.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "{{count}} Worker Node": "{{count}} Worker Node",
-  "{{count}} Worker Node_plural": "{{count}} Worker Nodes",
+  "{{count}} Worker Node_plural": "{{count}} Worker Node",
```
Comment (andrewballantyne) [inline dynamic-demo-plugin/locales/en/plugin__console-demo-plugin.json:3]: Not sure what is going on with the latest i18n stuff -- but was this intended to be changed in your PR? Seems unrelated.

**Review by andrewballantyne — COMMENTED** [review]
One inquiry, but otherwise the code looks good. I'll just spin up the plugin and give it a test before approving.

File: dynamic-demo-plugin/locales/en/plugin__console-demo-plugin.json (hunk 1 above)
Comment (jcaianirh) [reply dynamic-demo-plugin/locales/en/plugin__console-demo-plugin.json:3]: @andrewballantyne hmm interesting. I did run 'yarn i18n' I wonder if this was part of the output of that command...i can def remove that for now though.

Comment (andrewballantyne) [reply dynamic-demo-plugin/locales/en/plugin__console-demo-plugin.json:3]: Ah -- if this was done by the i18n tool -- not sure you'll be able to remove it. Must be a latent change from the i18n upgrade.

**Review by andrewballantyne — APPROVED** [review]

**Comment by andrewballantyne** [comment]
Looks good, addresses what I set out to do have done with the ticket. 👍

**Comment by jcaianirh** [comment]
QE Approver:
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

LGTM Approver:
/assign @jhadvig

**Review by jhadvig — APPROVED** [review]
/lgtm

File: dynamic-demo-plugin/console-extensions.json [hunk 2]
```diff
@@ -319,6 +319,16 @@
       "component": { "$codeRef": "overviewDetailItem.default" }
     }
   },
+  {
```
Comment (yapei) [inline dynamic-demo-plugin/console-extensions.json:322]: shall we remove `console.dashboards/overview/detail/item"` code at line 316 ~ line 319?

**Comment by yapei** [comment]
`console.dashboards/custom/overview/detail/item` extension is working, customized `Custom Overview Detail Title` item is correctly rendered on Details card, also checked customized detail item also can be rendered correctly when only required params set
Screen Shot 2022-09-21 at 14 35 42
